File: scala-dci/src/example/Application.java
package example;

import example.domain.Account;
import example.usecase.WithdrawalContext;

public class Application {

	public static void main(String[] args) throws Exception {
		Account savingAccount = new Account(50);
		Account checkingAccount = new Account(200);
		
		System.out.printf("Before Transaction: Savings %s, Checkings %s\n", savingAccount.getBalance(),
				checkingAccount.getBalance());
		
		WithdrawalContext context = new WithdrawalContext(checkingAccount, savingAccount, 100);
		context.execute();
		
		System.out.printf("After Transaction: Savings %s, Checkings %s\n", savingAccount.getBalance(),
				checkingAccount.getBalance());
	}

}

File: scala-dci/src/example/Application.scala
package example;

import example.domain.Account
import example.usecase.WithdrawalContext
import example.domain.SavingsAccount
import example.domain.CheckingAccount

object Application extends App {

		val savingAccount = new SavingsAccount(50.0)
		val checkingAccount = new CheckingAccount(200.0)
		
		println(s"Before Transaction: Savings $savingAccount, Checkings $checkingAccount\n")
		
//		val context = new WithdrawalContext(checkingAccount, savingAccount, 100);
//		context.execute();
		
		println(s"After Transaction: Savings $savingAccount, Checkings $checkingAccount\n")

}
